design/baremetal-operator/sub-states.md

```diff
@@ -0,0 +1,185 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# BareMetalHost detailed activity
+
+## Status
+
+implementable
+
+## Summary
+
+This design proposes a new field to specify more precisely at which point in
+a BMH lifecycle it is.
+
+## Motivation
+
+The primary motivation is to enable operators an easier insight into what
+is going on with their nodes. This simplifies debugging and makes people feel
+better about long-running processes. It may also be helpful for UI tools
+to be able to display some sort of progress information.
+
+### Goals
+
+* Provide a way to know what action is being executed right now without
+  accessing Ironic API.
+* Provide a rough estimate of the progress of execution.
+
+### Non-Goals
+
+* Provide access to the Ironic Node `provision_state` field.
+* Establish a direct one-to-one mapping between provision states and the
+  new field values.
+* Provide more granularity than what the Ironic API can report.
+* The values of the new field will not be a part of the API contract and can
+  change as a result of backend changes.
+* Define a non-empty detailed activity for each possible situation.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an operator reasonably familiar with the provisioning process, I would like
+to know what action is being executed on a BMH right now.
+
+## Design Details
+
+* The `ProvisionStatus` structure is extended with a new string field
+  `CurrentActivity` and a new integer field `Progress`.
+* The `Progress` value is a number from 0 to 100 that corresponds to estimated
```

Comment:
Do we think the time at which the step began/finished could be helpful for debugging? See design concept 2 of what this might look like in a GUI.

Reply:
I think it's potentially useful, although I have concerns about concept 2 (see my other message). However, I'd prefer to limit the scope of work for this design, because there's truly no limit to improving it :) Similarly, I'd start with an easier UI concept.